apps/champions/lib/champions/battle/simulator.ex

```diff
@@ -1088,6 +1111,110 @@ defmodule Champions.Battle.Simulator do
+         target,
+         history,
+         _target_initial_state
+       ) do
+    # If the execution is over, we remove it from the target
+    new_target =
+      update_in(target, [:executions_over_time], fn current_executions ->
+        Enum.filter(current_executions, fn exec -> exec.skill_id != execution_over_time.skill_id end)
+      end)
+
+    {new_target, history}
+  end
+
+  defp process_execution_over_time(
+         %{execution: %{"type" => "DealDamageOverTime"}, remaining_interval_steps: remaining_interval_steps} =
+           execution_over_time,
+         target,
+         history,
+         target_initial_state
+       ) do
+    if remaining_interval_steps == 0 do
+      apply_deal_damage_over_time(
+        execution_over_time,
+        target,
+        history,
+        target_initial_state
+      )
+    else
+      execution = target.executions_over_time |> Enum.find(fn exec -> exec == execution_over_time end)
+
+      new_execution_over_time =
+        Map.put(execution_over_time, :remaining_interval_steps, execution.remaining_interval_steps - 1)
+
+      new_target =
+        update_in(target, [:executions_over_time], fn current_executions ->
+          Enum.filter(current_executions, fn exec -> exec != execution end) ++ [new_execution_over_time]
+        end)
+
+      Logger.info("Remaining period: #{new_execution_over_time.remaining_interval_steps}")
+
+      {new_target, history}
+    end
+  end
+
+  defp process_execution_over_time(
+         execution_over_time,
+         target,
+         history,
+         _target_initial_state
+       ) do
+    Logger.warning(
+      "#{format_unit_name(execution_over_time.caster)} tried to apply an unknown execution over time to #{format_unit_name(target)}"
+    )
+
+    {target, history}
+  end
+
+  defp apply_deal_damage_over_time(execution_over_time, target, history, target_initial_state) do
+    damage_after_defense =
+      calculate_damage(execution_over_time.caster, target_initial_state, execution_over_time.execution["attack_ratio"])
+
+    Logger.info(
+      "#{format_unit_name(execution_over_time.caster)} dealing #{damage_after_defense} damage to #{format_unit_name(target)} (#{target.health} -> #{target.health - damage_after_defense}). Steps remaining: #{execution_over_time.remaining_duration}."
+    )
+
+    new_history =
+      add_to_history(
+        history,
+        %{
+          target_id: target.id,
+          skill_id: execution_over_time.skill_id,
+          stat_affected: %{stat: :HEALTH, amount: -damage_after_defense}
+        },
+        :execution_received
+      )
+
+    initial_interval_steps = get_interval_steps(execution_over_time.execution)
+
+    new_target =
+      target
+      |> Map.put(:health, target.health - damage_after_defense)
+      |> update_in([:executions_over_time], fn current_executions ->
+        Enum.map(current_executions, fn exec ->
+          if exec == execution_over_time do
```

Comment:
I think there is a really, really edge-y edge case here where this fails still. Do you think we should add an issue to revisit this later?

Reply:
As discussed, the case in which two units of the same character apply the same EoT to a target would be handled by this comparison because the execution over time would have different values in the `caster` field.